Cargo console ID check

Nukeuler123 opened the pull request:
### Purpose
Fixes #4242 
Allows the cargo console to check for specific IDs. (QM, Caption, HOP, CargoTech, NULL, AI, Cyborg)

### Notes:
In Pogstation there is a custom gameobject called Cargo Console instead of supply console, it has been left alone for mappers to fix and in case changing a component on it breaks something. Will make issue.

Commit:
Cargo console ID check

Allows the cargo console to check for specific IDs. (QM, Caption, HOP, CargoTech, NULL, AI, Cyborg)

corp-0 commented on UnityProject/Assets/Scripts/Consoles/CargoConsole.cs:
```diff
@@ -0,0 +1,61 @@
+	private GUI_Cargo associatedTab;
+
+	[SerializeField]
+	private List<JobType> allowedTypes = new List<JobType>();
```
```suggestion
	private List<JobType> allowedTypes = null;
```

corp-0 commented on UnityProject/Assets/Scripts/Consoles/CargoConsole.cs:
```diff
@@ -0,0 +1,61 @@
+
+	public bool WillInteract(HandApply interaction, NetworkSide side)
+	{
+		if (!DefaultWillInteract.Default(interaction, side))
```
```suggestion
		return DefaultWillInteract.Default(interaction, side) &&
		       Validations.HasComponent<IDCard>(interaction.HandObject);
```

corp-0 commented on UnityProject/Assets/Scripts/Consoles/CargoConsole.cs:
```diff
@@ -0,0 +1,61 @@
+	public bool WillInteract(HandApply interaction, NetworkSide side)
+	{
+		if (!DefaultWillInteract.Default(interaction, side))
+			return false;
```
```suggestion
```

corp-0 replied:
```suggestion
```

corp-0 commented on UnityProject/Assets/Scripts/Consoles/CargoConsole.cs:
```diff
@@ -0,0 +1,61 @@
+	{
+		if (!DefaultWillInteract.Default(interaction, side))
+			return false;
+
```
```suggestion
```

corp-0 commented on UnityProject/Assets/Scripts/Consoles/CargoConsole.cs:
```diff
@@ -0,0 +1,61 @@
+		if (!DefaultWillInteract.Default(interaction, side))
+			return false;
+
+		//interaction only works if using an ID card on console
```
```suggestion
```

corp-0 commented on UnityProject/Assets/Scripts/Consoles/CargoConsole.cs:
```diff
@@ -0,0 +1,61 @@
+			return false;
+
+		//interaction only works if using an ID card on console
+		if (!Validations.HasComponent<IDCard>(interaction.HandObject))
```
```suggestion
```

corp-0 commented on UnityProject/Assets/Scripts/Consoles/CargoConsole.cs:
```diff
@@ -0,0 +1,61 @@
+
+		//interaction only works if using an ID card on console
+		if (!Validations.HasComponent<IDCard>(interaction.HandObject))
+			return false;
```
```suggestion
```

corp-0 commented on UnityProject/Assets/Scripts/Consoles/CargoConsole.cs:
```diff
@@ -0,0 +1,61 @@
+		//interaction only works if using an ID card on console
+		if (!Validations.HasComponent<IDCard>(interaction.HandObject))
+			return false;
+
```
```suggestion
```

corp-0 commented on UnityProject/Assets/Scripts/Consoles/CargoConsole.cs:
```diff
@@ -0,0 +1,61 @@
+		if (!Validations.HasComponent<IDCard>(interaction.HandObject))
+			return false;
+
+		return true;
```
```suggestion
```

corp-0 reviewed (commented):
just suggestions, its on you if you want to apply these

Commit:
Apply suggestions from code review

Co-authored-by: corp0 <43683714+corp-0@users.noreply.github.com>

Commit:
Bugfix

Made it so that an action message is displayed when  you swipe your card and fixed a bug with the text not updating, it aint elegant but it works

Commit:
Merge remote-tracking branch 'origin/bug-fixes' into bug-fixes

corp-0 commented on UnityProject/Assets/Scripts/Consoles/CargoConsole.cs:
```diff
@@ -0,0 +1,75 @@
+		}
+		else
+		{
+			Chat.AddActionMsgToChat(playeref, "You swipe your ID through the supply console's ID slot, t" +
```
Break the string before the “t”, so the next string has the entire word

corp-0 replied:
```suggestion
			Chat.AddActionMsgToChat(playeref, "You swipe your ID through the supply console's ID slot, " +
```

corp-0 commented on UnityProject/Assets/Scripts/Consoles/CargoConsole.cs:
```diff
@@ -0,0 +1,75 @@
+		{
+			Chat.AddActionMsgToChat(playeref, "You swipe your ID through the supply console's ID slot, t" +
+			                                  "he console denies your ID",
+				"swiped their ID through the supply console's ID slot");
```
Missing the character name here. You can format the string like this
```
$”{playeref.ExpensiveName()} swiped their ID blah blah blah”
```
Also you should be able to get the possessive of the character’s gender but I’m not quite sure where is the function at the moment.

corp-0 replied:
```suggestion
				$"{playeref.ExpensiveName()} swiped their ID through the supply console's ID slot");
```

corp-0 reviewed (commented):
Just gave it a quick look since I’m already in bed. Will test tomorrow!

corp-0 commented on UnityProject/Assets/Scripts/Consoles/CargoConsole.cs:
```diff
@@ -0,0 +1,75 @@
+		else
+		{
+			Chat.AddActionMsgToChat(playeref, "You swipe your ID through the supply console's ID slot, t" +
+			                                  "he console denies your ID",
```
```suggestion
			                                  "the console denies your ID",
```

corp-0 committed:
Update UnityProject/Assets/Scripts/Consoles/CargoConsole.cs

corp-0 committed:
Update UnityProject/Assets/Scripts/Consoles/CargoConsole.cs

corp-0 committed:
Update UnityProject/Assets/Scripts/Consoles/CargoConsole.cs
